src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/DelegateTests.cs

```diff
@@ -195,5 +195,233 @@ public static void InvokeFunctionAcceptingArrayTypes(Function objectPrototype, s
+                dispatcher.Invoke("fireEvent", evnti);
+                Runtime.InvokeJS("gc();");// needs v8 flag --expose-gc
+            }
+        }
+
+        [Fact]
+        public static void NullDelegate()
+        {
+            var tcs = new TaskCompletionSource<int>();
+            var factory = new Function("delegate", "callback", @"
+                callback(delegate);
+            ");
+
+            Delegate check = null;
+            Action<Delegate> callback = (Delegate data) =>
+            {
+                check = data;
+            };
+            Assert.Null(check);
+            factory.Call(null, null, callback);
+            Assert.Null(check);
+        }
+
+        [Fact]
+        public static async Task ResolveStringPromise()
+        {
+            var factory = new Function(@"
+                return new Promise((resolve, reject) => {
+                  setTimeout(() => {
+                    resolve('foo');
+                  }, 300);
+                });");
+
+            var promise = (Task<object>)factory.Call();
+            var value = await promise;
+
+            Assert.Equal("foo", (string)value);
+            
+        }
+
+        [Fact]
+        public static async Task ResolveJSObjectPromise()
+        {
+            for (int i = 0; i < 10; i++)
+            {
+                var factory = new Function(@"
+                return new Promise((resolve, reject) => {
+                  setTimeout(() => {
+                    resolve({foo:'bar'});
+                  }, 300);
+                });");
+
+                var promise = (Task<object>)factory.Call();
+                var value = (JSObject)await promise;
+
+                Assert.Equal("bar", value.GetObjectProperty("foo"));
+
+                Runtime.InvokeJS("gc();");// needs v8 flag --expose-gc
+            }
+        }
+
+        [Fact]
+        public static async Task RejectPromise()
+        {
+            var factory = new Function(@"
+                return new Promise((resolve, reject) => {
+                  setTimeout(() => {
+                    reject('fail');
+                  }, 300);
+                });");
+
+            var promise = (Task<object>)factory.Call();
+
+            var ex = await Assert.ThrowsAsync<JSException>(async () => await promise);
+            Assert.Equal("fail", ex.Message);
+        }
+
+        [Fact]
+        public static async Task RejectPromiseError()
+        {
+            var factory = new Function(@"
+                return new Promise((resolve, reject) => {
+                  setTimeout(() => {
+                    reject(new Error('fail'));
+                  }, 300);
+                });");
+
+            var promise = (Task<object>)factory.Call();
+
+            var ex = await Assert.ThrowsAsync<JSException>(async () => await promise);
+            Assert.Equal("Error: fail", ex.Message);
+        }
+
+
+        [ActiveIssue("not implemented")]
```

Comment:
I expect we have a convention for what to do here, maybe create a dummy tracking issue for todo items before this PR is committed

Reply:
https://github.com/dotnet/runtime/issues/56963